### PR TITLE
next: Fix #74258

### LINKED ISSUE #74258: gsettings-desktop-schemas dep missing for Next browser
## Issue description

https://github.com/atlas-engineer/next/issues/496 says that Next only renders a blank page on Nix.  In my experience, this is because the propagated input `gsettings-desktop-schemas` is missing.

### Steps to reproduce

Install Next in a profile that does not have `gsettings-desktop-schemas`.  Start Next.

## Technical details

Reporting this on behalf of a Nix user, I don't have Nix myself, sorry.

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Commit message
Next had a few issues with its packaging:
* the platform port was exposed in all-packages
  And this is not useful for outside users.
  It's now a local attribute in the next package.

* the platform port wasn't wrapped correctly
  It appears that the lisp core was being wrapped,
  when instead the actual gtk application that's
  called within the lisp core had to be wrapped.

* codestyle/indentation

###### Description of change
This is meant to fix #74258.

According to the Next documentation at https://github.com/atlas-engineer/next/tree/master/ports#next-platform-ports, the only part of Next that needs to be treated like a gtk app is the webkitgtk platform port.
To fix this issue we just wrapped this program like any other gtk app, as pre the documentation
* https://nixos.org/nixpkgs/manual/#sec-language-gnome

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/browsers/next/next-gtk-webkit.nix [hunk 1]
```diff
@@ -1,20 +1,43 @@
...
+    gst_all_1.gst-plugins-base
+    gst_all_1.gstreamer
   ];
+
+  makeFlags = [
+    "gtk-webkit"
+    "PREFIX=${placeholder "out"}"
+  ];
+
+  installTargets = "install-gtk-webkit";
```
Comment (jtojnar) [inline pkgs/applications/networking/browsers/next/next-gtk-webkit.nix]: Structured attrs:

```suggestion
  installTargets = [ "install-gtk-webkit" ];
```

File: pkgs/applications/networking/browsers/next/default.nix [hunk 2]
```diff
@@ -1,86 +1,90 @@
...
-    meta = with stdenv.lib; {
-      description = "Infinitely extensible web-browser (with Lisp development files)";
-      homepage = https://next.atlas.engineer;
-      license = licenses.bsd3;
-      maintainers = [ maintainers.lewo ];
-      platforms = [ "x86_64-linux" ];
-    };
-  }
+  meta = with stdenv.lib; {
+    description = "Infinitely extensible web-browser (with Lisp development files using webkitgtk platform port)";
```
Comment (jtojnar) [inline pkgs/applications/networking/browsers/next/default.nix]:
```suggestion
    description = "Infinitely extensible web-browser (with Lisp development files using WebKitGTK platform port)";
```

**Review by jonringer — COMMENTED** [review]
seems to hang when trying to go to `google.com`

**Comment by worldofpeace** [comment]
In a pure nix-shell I see the following in the logs
`/home/worldofpeace/.local/share/next/next-gtk-webkit.log`
```
Failed to load TLS database: Failed to load system trust store: Error while reading file.
```

when visiting `google.com`. Not other websites though.

As this has been reported upstream, I'm going to assume it's a bug
* https://github.com/atlas-engineer/next/issues/296

**Comment by jonringer** [comment]
if it's an upstream issue, then I'm fine with the change :)

**Comment by nlewo** [comment]
Just note I exposed the `next-browser-webkit` package to allow users to override it.
But I'm not opposed to the current change.

@worldofpeace Are you using a formatting tool?

Otherwise, I locally tested this PR which works well!

**Comment by worldofpeace** [comment]
> Just note I exposed the `next-browser-webkit` package to allow users to override it.
> But I'm not opposed to the current change.
> 
> @worldofpeace Are you using a formatting tool?
> 
> Otherwise, I locally tested this PR which works well!

Yeah, `nixpkgs-fmt` and some of my personal stylistic opinions https://discourse.nixos.org/t/document-attribute-ordering-in-package-expressions/4887.

**Comment by worldofpeace** [comment]
> > Just note I exposed the `next-browser-webkit` package to allow users to override it.
> But I'm not opposed to the current change.

I'm not sure why it has be separate, and not like all-in-one in the build with some sort of flag.

File: pkgs/applications/networking/browsers/next/next-gtk-webkit.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    glib-networking
+    gsettings-desktop-schemas
+    gtk3
+    webkitgtk
+    gst_all_1.gst-plugins-base
+    gst_all_1.gstreamer
+  ];
+
+  makeFlags = [
+    "gtk-webkit"
```
Comment (jtojnar) [inline pkgs/applications/networking/browsers/next/next-gtk-webkit.nix]: This should be in `buildFlags`.

**Comment by jtojnar** [comment]
I tried

```patch
diff --git a/pkgs/applications/networking/browsers/next/default.nix b/pkgs/applications/networking/browsers/next/default.nix
index 97c65fc12ef..6cc5316c098 100644
--- a/pkgs/applications/networking/browsers/next/default.nix
+++ b/pkgs/applications/networking/browsers/next/default.nix
@@ -3,6 +3,8 @@
 , lispPackages
 , sbcl
 , callPackage
+, libfixposix
+, xclip
 }:
 
 let
@@ -59,6 +61,7 @@ stdenv.mkDerivation rec {
     trivia
     trivial-clipboard
     unix-opts
+    quicklisp
   ];
 
   prePatch = ''
@@ -66,16 +69,23 @@ stdenv.mkDerivation rec {
       --replace "next-gtk-webkit" "${next-gtk-webkit}/bin/next-gtk-webkit"
   '';
 
-  buildPhase = ''
-    common-lisp.sh --eval "(require :asdf)" \
-                   --eval "(asdf:load-asd (truename \"next.asd\") :name \"next\")" \
-                   --eval '(asdf:make :next)' \
-                   --quit
+  preConfigure = ''
+    export HOME=$TMPDIR
   '';
 
-  installPhase = ''
-    install -D -m0755 next $out/bin/next
-  '';
+  makeFlags = [
+    "PREFIX=${placeholder ''out''}"
+    "NEXT_INTERNAL_QUICKLISP=false"
+  ];
+
+  buildFlags = [
+    "next"
+  ];
+
+  installTargets = [
+    "install-next"
+    "install-assets"
+  ];
 
   # Stripping destroys the generated SBCL image
   dontStrip = true;
```

to see if it fixes https://github.com/NixOS/nixpkgs/pull/76838 but it failed with

```
  Package QL does not exist.
```

**Pull request merged**